### PR TITLE
Add the possibility to get the number of remaining messages

### PR DESCRIPTION
**Pull request by Snowad14**
I don't know if I implemented it the way you want but I personally needed it and maybe other people did too

### CODE REVIEW

File: poe-api/src/poe.py [hunk 1]
```diff
@@ -147,6 +147,16 @@ def get_bot_names(self):
       bot_names[bot_nickname] = bot_obj["displayName"]
     return bot_names
   
+  def get_remaining_messages(self, chatbot):
+    url = f'https://poe.com/_next/data/{self.next_data["buildId"]}/{chatbot}.json'
+    limited_chatbots = ["GPT-4", "Claude+"]
+    if chatbot in limited_chatbots:
```
Comment (ading2210) [inline poe-api/src/poe.py:153]: There's no need to make the request manually when there's already a `client.get_bot` function for this.

File: poe-api/src/poe.py [hunk 2]
```diff
@@ -147,6 +147,16 @@ def get_bot_names(self):
       bot_names[bot_nickname] = bot_obj["displayName"]
     return bot_names
   
+  def get_remaining_messages(self, chatbot):
+    url = f'https://poe.com/_next/data/{self.next_data["buildId"]}/{chatbot}.json'
+    limited_chatbots = ["GPT-4", "Claude+"]
```
Comment (ading2210) [inline poe-api/src/poe.py:152]: There's no need to hardcode the bot names, that'll just cause the function to break whenever Poe changes their bots.

File: poe-api/src/poe.py [hunk 3]
```diff
@@ -147,6 +147,16 @@ def get_bot_names(self):
...
+    if chatbot in limited_chatbots:
+      r = request_with_retries(self.session.get, url)
+      remaining_count = r.json()["pageProps"]["payload"]["chatOfBotDisplayName"]["defaultBotObject"]["messageLimit"]["numMessagesRemaining"]
+      return remaining_count
+    else:
+      raise Exception(f"Chatbot {chatbot} is not limited or doesn't exist.")
```
Comment (ading2210) [inline poe-api/src/poe.py:158]: Maybe don't raise an exception for if the chatbot doesn't have a limit. The `numMessagesRemaining` value is present on all chatbots anyways, and it'll just be `null` if there isn't a limit.

**Review by ading2210 — CHANGES_REQUESTED** [review]

File: poe-api/src/poe.py [hunk 4]
```diff
@@ -147,6 +147,10 @@ def get_bot_names(self):
       bot_names[bot_nickname] = bot_obj["displayName"]
     return bot_names
   
+  def get_remaining_messages(self, chatbot):
+    chat_data = self.get_bot(chatbot)
```
Comment (ading2210) [inline poe-api/src/poe.py:151]: `client.get_bot` takes the bot display name as an input, so you'd have to do `self.get_bot(self.bot_names[chatbot])`

**Review by ading2210 — CHANGES_REQUESTED** [review]

File: poe-api/src/poe.py (hunk 4 above)
Comment (Snowad14) [reply poe-api/src/poe.py]: I used to use the name "GPT-4" directly but it's true that using the official code names is better

**Pull request merged**